client/trino-client/src/main/java/io/trino/client/auth/external/HttpTokenPoller.java

```diff
@@ -72,7 +73,33 @@ public TokenPollResult pollForToken(URI tokenUri, Duration timeout)
         }
     }
 
-    private static Request createRequest(URI tokenUri)
+    @Override
+    public void tokenReceived(URI tokenUri)
+    {
+        try {
+            Failsafe.with(new RetryPolicy<Integer>()
+                    .withMaxAttempts(11)
```

Comment:
why -11? I find `withMaxDuration` more readable.

Reply:
Ok so here the main concern here is to try as best as we can, but not too hard. For cases like this I feel that having a strict max attempts limit is a bit better. kick-back is added to allow server to recover in case of emergency, but if after 10 retries it still is not ready to accept our request, then we should just give up. We can change it to MaxDuration, but then one would have to calculate in their head how many times will the client retry to communicate before the call will fail.